sphinx/tutorial/cylc/furthertopics/retries.rst

```diff
@@ -0,0 +1,161 @@
+
+Create a new suite by running the following commands::
+
+   rose tutorial retries-tutorial
+   cd retries-tutorial
+
+You will now have a suite with a ``roll_doubles`` task which simulates
+trying to roll doubles using two dice:
+
+.. code-block:: cylc
+
+   [cylc]
+       UTC mode = True # Ignore DST
+
+   [scheduling]
+       [[dependencies]]
+           graph = start => roll_doubles => win
+
+   [runtime]
+       [[start]]
+       [[win]]
+       [[roll_doubles]]
+           script = """
+   sleep 10
+   RANDOM=$$  # Seed $RANDOM
+   DIE_1=$((RANDOM%6 + 1))
+   DIE_2=$((RANDOM%6 + 1))
+   echo "Rolled $DIE_1 and $DIE_2..."
+   if (($DIE_1 == $DIE_2)); then
+      echo "doubles!"
+   else
+      exit 1
+   fi
+   """
+
+
+Running Without Retries
+-----------------------
+
+Let's see what happens when we run the suite as it is. Open the cylc gui::
+
+   cylc gui retries-tutorial &
+
+Then run the suite::
+
+   cylc run retries-tutorial
+
+Unless you're lucky, the suite should fail at the roll_doubles task.
+
+Stop the suite::
+
+   cylc stop retries-tutorial
+
+
+Configuring Retries
+-------------------
+
+We need to tell cylc to retry it a few times - replace the line
+``[[roll_doubles]]`` in the ``suite.rc`` file with:
+
+.. code-block:: cylc
+
+   [[roll_doubles]]
+       [[[job]]]
+           execution retry delays = 5*PT6S
+
+This means that if the ``roll_doubles`` task fails, cylc expects to
+retry running it 5 times before finally failing. Each retry will have
+a delay of 6 seconds.
+
+We can multiple retry periods with the ``execution retry delays`` setting by
+separating them with commas, for example the following line would tell cylc to
+retry a task, first after 15 seconds, then after 10 minutes, then after one hour
+then after three hours.
+
+.. code-block:: cylc
+
+   execution retry delays = PT15S, PT10M, PT1H, PT3H
+
+We've chosen 6 seconds because it's relatively easy to observe for this example.
+
+
+Running With Retries
+--------------------
+
+If you closed it, re-open the cylc gui::
+
+   cylc gui retries-tutorial &
+
+Re-run the suite::
+
+   cylc run retries-tutorial
+
+What you should see is cylc retrying the ``roll_doubles`` task. Hopefully,
+it will succeed (about a 1 in 3 chance of every task failing) and the
+suite will continue.
+
+
+Altering Behaviour
+------------------
+
+We can alter the behaviour of the task based on the number of retries, using
+``$CYLC_TASK_TRY_NUMBER``.
+
+Change the ``script`` setting for the ``roll_doubles`` task to this::
+
+   sleep 10
+   RANDOM=$$  # Seed $RANDOM
+   DIE_1=$((RANDOM%6 + 1))
+   DIE_2=$((RANDOM%6 + 1))
+   echo "Rolled $DIE_1 and $DIE_2..."
+   if (($DIE_1 == $DIE_2)); then
+       echo "doubles!"
+   elif (($CYLC_TASK_TRY_NUMBER >= 2)); then
+       echo "look over there! ..."
+       echo "doubles!"  # Cheat!
+   else
+       exit 1
+   fi
+
+If your suite is still running, stop it, then run it again::
+
+   cylc run retries-tutorial
```

Comment:
We don't need to tell the reader how to run their suite for a third time! Even the second inclusion of this same command is unnecessary, but at least drop this one.